File: Bird and Max Fruit Gathering.java
// Problem: Bird and Max Fruit Gathering
// geeksforgeeks problem of the day -> 4th September 2026
// JAVA CODE
class Solution {
    public int maxFruits(ArrayList<Integer> arr, int m) {
        int n = arr.size();
        
        // We can visit all trees
        if(m >= n){
            int total = 0;
            for(int x:arr){
                total+=x;
            }
            return total;
        }

        // First window
        int sum = 0;
        for(int i=0;i<m;i++){
            sum+= arr.get(i);
        }
        
        int ans = sum;
        
        // Remaining n - 1 windows
        for(int window = 1;window<n;window++){
            int sub = window - 1;
            sum-= arr.get(sub);
            
            int add = (window + m - 1)%n;
            sum+= arr.get(add);
            
            ans = Math.max(ans, sum);
        }
        return ans;
    }
}

File: Bird and Max Fruit Gathering.cpp
// Problem: Bird and Max Fruit Gathering
// geeksforgeeks problem of the day -> 4th September 2026
// C++ CODE
class Solution {
  public:
    int maxFruits(vector<int>& arr, int m) {
        int ans = 0, n = arr.size();
        int temp = 0;
        for(int i=0; i<(n+m); i++) {
            if(i < m) {
                ans += arr[i];
                temp = ans;
            } else {
                temp += arr[i % n];
                temp -= arr[(i-m) % n];
                ans = max(ans, temp);
            }
        }
        return ans;
    }
};